### PR TITLE
Skip recycling quest on elements with "note" tag, as those are likely to be problematic 

### PR DESCRIPTION
**Pull request by mnalis**
Recycling quest should be skipped on elements having `note=* `tag (as its existence likely indicates the issue is too complicated for AI, and human is required to evaluate specific issues detailed in freeform text of `note`)

(see https://github.com/streetcomplete/StreetComplete/issues/4303#issuecomment-1507617359 and around for  context)

(Tested in https://github.com/mnalis/StreetComplete/actions/runs/4737627521 on e.g. [n2486015492](https://www.openstreetmap.org/node/2486015492), [n2486015499](https://www.openstreetmap.org/node/2486015499) and others)

### CODE REVIEW

**Comment by westnordost** [comment]
>  note=* tag (as its existence likely indicates the issue is too complicated for AI, and human is required to evaluate specific issues detailed in freeform text of note)

Wouldn't this apply to any element? Or maybe not, maybe the note could also inform about that the feature isn't visible yet on aerial imagery? A note could include all sorts of stuff, relevant or not relevant.
If someone is on-site however, most if not all relevancy a note might have is void, because the mapper is checking the ground truth on site with his own eyes.

**Comment by mnalis** [comment]
TL;DR: it would make this quest **safer** by avoiding **potentially** problematic recycling containers, while **retaining its full usefulness** (as only a very small percentage of objects would be skipped).

---

> Wouldn't this apply to any element? Or maybe not, maybe the note could also inform about that the feature isn't visible yet on aerial imagery? A note could include all sorts of stuff, relevant or not relevant.

Agreed, it does depends on the note.  It is still a huge indicator that human review of the note is needed. However, as there were only about `1 400` of such combinations (of `recycling_type=container` + `*note*=*`) worldwide, it seemed to me:
- that it would be playing it safe by avoiding such _**potentially** problematic_  elements
- that it would not seriously reduce usefulness of this quest, as in vast majority of the cases it would continue to work as it did before

> If someone is on-site however, most if not all relevancy a note might have is void, because the mapper is checking the ground truth on site with his own eyes.

That is a possibility. But to be sure, one would have to display such `note` to the user, and ask for their feedback if that is now resolved (so SC can remove `note` tag),  or if not, if they want to edit such `note` to add more information they now know as a on-ground surveyor. Such advanced feature would however be:

- somewhat of an overkill given relatively low number of such elements (now, if say 30% of recycling containers had such notes, _or_ if it would considered to apply such feature to **all quests**, it might be worthwhile!)
- maybe not fully in line with SC "be simple" philosophy
- determining whether it would be applicable to all elements would be much bigger task in itself (compared to just checking it in `recycling_type=container` case where it was shown to be the problem in linked issue)
- beyond my capabilities to implement anyway

So I've rather tried to keep it safe and simple instead.

**Comment by westnordost** [comment]
A `note` is not a `fixme`. A `note` could stay forever and also not even hint at a problem, i.e. there is absolutely no necessity to resolve it. If it was a problem, it was a `fixme`.

I get where you are coming from, but my point of view is that:
- it would be inconsistent to only ignore all things tagged with anything `*note*` for this specific quest
- see my last comment, only because **potentially** the presence of a `note` could mean that in one way or another, the data on which the SC quest is based on does not apply is IMO too weak of a reason to generally ignore anything that has a `note`
- You are basically saying that anything `note` should be interpreted as "*do not edit unless you read this!*" and I don't really subscribe to this interpretation. If this interpretation is going to be established, it will just be another thing that makes editing the map somewhat more complicated, being up to the "experts"
- finally, I remember the issue that spawned this PR. And I have to say, the case that was explained (location of separately mapped containers mapped very closely to each other changes regularly), ... well, I see the issue more with how it was mapped, i.e. that they are mapped as separate containers in the first place even though they change/switch location regularly.

**Comment by mnalis** [comment]
> A note is not a fixme. A note could stay forever and also not even hint at a problem, i.e. there is absolutely no necessity to resolve it. 

Yes. But note for example two nodes in example at the top: [n2486015492](https://www.openstreetmap.org/node/2486015492), [n2486015499](https://www.openstreetmap.org/node/2486015499) - those are two nearby containers, and note (in Italian) details them by color. 

If user solving the SC quest was asked for those containers, there is 50% chance that they'd answer the wrong one, thus making that `note` not only "not helpful", but actively incorrect / confusing, creating more problems down the line for next mappers. So it would be better (if far less than ideal) if that `note` was removed in those cases.

So perhaps to fix those we should just remove the `note` when we answer the quest? But that would break other cases like [n4537681043](https://www.openstreetmap.org/node/4537681043) or [n7098727064](https://www.openstreetmap.org/node/7098727064) where the `note` should stay regardless of how the quest was answered.

> If it was a problem, it was a fixme. (Uh, are we ignoring anything fixme? Should we?)

My opionion was (and is) that we should be fixing them instead :smile: e.g.:
- https://github.com/streetcomplete/StreetComplete/issues/3084

(but that one has been implemented in [SCEE](https://github.com/Helium314/SCEE) after being rejected here)

> it would be inconsistent to only ignore all things tagged with anything *note* for this specific quest

We're already looking at notes in other quest like AddPathSurface, AddPitchSurface, AddRoadSurface, AddCyclewayPartSurface, AddFootwayPartSurface... 

And as explained, **I've opted to implement it only for this quest because cost/benefit is clear here**. If I suggested to deploy that it in all 150+ quests, that would probably require me to do hundred and fifty times more cost/benefit analyses for all of them (and like here without guarantee it would not be for nought), and the issues have not been brought up for other quests yet where it would help (while this is at least second issue with recycling containers IIRC), so one might not have much of the justification to invest all that effort.

> see my last comment, only because potentially the presence of a note could mean that in one way or another, the data on which the SC quest is based on does not apply is IMO too weak of a reason to generally ignore anything that has a note

I'm aware. As you probably too remember (from that [KEYS_THAT_SHOULD_BE_REMOVED_WHEN_SHOP_IS_REPLACED](https://github.com/mnalis/StreetComplete-taginfo-categorize) episode) that I'm _**much more opposed to damaging (even small amounts of) data by negligence**_ and would much prefer not to change some part of the map at all, than to (even **if just potentially**) damage it instead of improving it.

Especially if the **cost of avoiding such damage is so tiny** (so the few thousand recycling containers won't be having a SC quest)

> You are basically saying that anything note should be interpreted as "do not edit unless you read this!" and I don't really subscribe to this interpretation. If this interpretation is going to be established, it will just be another thing that makes editing the map somewhat more complicated, being up to the "experts"

But would it really be so horrible **if SC was to show 1% less quests** in order to much reduce potential damage it might lead its users to unknowingly do (even if that damage is just a small percentage of all SC edits)? To me it seems that SC skips many quests that it could be asking (in fact it was one of main reasons for my own fork), so **SC should not really be so averse to reducing that number of quests asked by a tiny amount more for a good cause**.

But anyway, that was my pitch. If you decide to skip all quests with elements that have `note` globally,  great! If you decide to accept just this PR, plenty good. If you decide to close it, I'll (probably :smiling_face_with_tear:) survive too, with NHF.

**Comment by westnordost** [comment]
Any comments about this from other contributors?

**Comment by matkoniecz** [comment]
I actually considered at some point skipping all objects with `note` `fixme` etc where risk of unexpected or already broken data is higher.

Though it never went further than "maybe it is worth investigating is it going to be helpful".

> it would be inconsistent to only ignore all things tagged with anything *note* for this specific quest

I agree, either it should be skipped for all quests (and defined in one place), definitely not implemented by each quest on its own.

>  the issue that spawned this PR. And I have to say, the case that was explained (location of separately mapped containers mapped very closely to each other changes regularly), ... well, I see the issue more with how it was mapped, i.e. that they are mapped as separate containers in the first place even though they change/switch location regularly amongst each other.

In such case SC just revealed already present bad mapping. Mapping each container from group as own `amenity=recycling` is dubious, when container often exchange positions it is simply incorrect.

-------------------

On the other hand, some `note` objects are going to be legitimately tricky and SC has higher chance to damage them.

----------------

> You are basically saying that anything note should be interpreted as "do not edit unless you read this!" and I don't really subscribe to this interpretation. 

I think I at least partially subscribe to this interpretation, but I am not really sure.

-------------------

Overall, I still have no strong preferences but I am not planning on implementing it and I would be against doing it for a single quest.

**Comment by westnordost** [comment]
I will close this then. Sorry, @mnalis .
